cmd/start.go

```diff
@@ -218,964 +196,924 @@ func setupStartCommand() *cli.Command {
-
-	var remoteSupernodeExecFile string
-
-	remoteSuperNodeConfigFilePath = strings.ReplaceAll(remoteSuperNodeConfigFilePath, "\\", "/")
-	remoteSupernodeExecFile = filepath.Join(string(remotePastelExecPath), constants.SuperNodeExecName[constants.OSType(string(remoteOsType))])
-	remoteSupernodeExecFile = strings.ReplaceAll(remoteSupernodeExecFile, "\\", "/")
-
-	/*	client.Cmd(fmt.Sprintf("rm %s", remoteSuperNodeConfigFilePath)).Run()
-
-		client.Cmd(fmt.Sprintf("echo -e \"%s\" >> %s", configs.SupernodeYmlLine1, remoteSuperNodeConfigFilePath)).Run()
-		client.Cmd(fmt.Sprintf("echo -e \"%s\" >> %s", configs.SupernodeYmlLine2, remoteSuperNodeConfigFilePath)).Run()
-		client.Cmd(fmt.Sprintf("echo -e \"%s\" >> %s", configs.SupernodeYmlLine3, remoteSuperNodeConfigFilePath)).Run()
-		client.Cmd(fmt.Sprintf("echo -e \"%s\" >> %s", fmt.Sprintf(configs.SupernodeYmlLine4, pastelid), remoteSuperNodeConfigFilePath)).Run()
-		client.Cmd(fmt.Sprintf("echo -e \"%s\" >> %s", configs.SupernodeYmlLine5, remoteSuperNodeConfigFilePath)).Run()
-		client.Cmd(fmt.Sprintf("echo -e \"%s\" >> %s", configs.SupernodeYmlLine6, remoteSuperNodeConfigFilePath)).Run()
-		client.Cmd(fmt.Sprintf("echo -e \"%s\" >> %s", fmt.Sprintf(configs.SupernodeYmlLine7, extIP), remoteSuperNodeConfigFilePath)).Run()
-		client.Cmd(fmt.Sprintf("echo -e \"%s\" >> %s", fmt.Sprintf(configs.SupernodeYmlLine8, fmt.Sprintf("%d", flagMasterNodeRPCPort)), remoteSuperNodeConfigFilePath)).Run()
-	*/
-	time.Sleep(5000 * time.Millisecond)
-
-	log.WithContext(ctx).Infof("Remote:::Start supernode command : %s", fmt.Sprintf("%s %s", remoteSupernodeExecFile, fmt.Sprintf("--config-file=%s", remoteSuperNodeConfigFilePath)))
-
-	go client.Cmd(fmt.Sprintf("%s %s", remoteSupernodeExecFile, fmt.Sprintf("--config-file=%s", remoteSuperNodeConfigFilePath))).Run()
-
-	defer client.Close()
+	var conf map[string]interface{}
 
-	log.WithContext(ctx).Info("Remote:::Waiting for supernode started...")
-	time.Sleep(5000 * time.Millisecond)
+	json.Unmarshal([]byte(confFile), &conf)
 
-	log.WithContext(ctx).Info("Remote:::Supernode was started successfully")
-	return nil
-}
+	for k := range confData {
+		if conf[k] != nil {
+			confDataValue := confData[k].(map[string]string)
+			confValue := conf[k].(map[string]interface{})
+			for itemKey := range confDataValue {
+				if len(confDataValue[itemKey]) != 0 {
+					confValue[itemKey] = confDataValue[itemKey]
 
-func runPastelNode(ctx context.Context, config *configs.Config, reindex bool, interactive bool) (err error) {
-	var pastelDPath string
+				}
+			}
+		}
+	}
 
-	if _, pastelDPath, _, _, err = checkPastelInstallPath(ctx, config, ""); err != nil {
-		return errNotFoundPastelPath
+	var updatedConf []byte
+	if updatedConf, err = json.Marshal(conf); err != nil {
+		return false, err
 	}
 
-	if err = checkStartNodeParams(ctx, config); err != nil {
-		return err
+	if ioutil.WriteFile(masternodeConfPath, updatedConf, 0644) != nil {
+		return false, err
```

Comment:
it's better to wrap error with detail off error.
return false, errros.Errorf("failed to write masternode cofinig file: %v", err)